### PR TITLE
Train final nf1 model

### PR DESCRIPTION
**Pull request by MattsonCam**
In this pr, the final nf1 model was trained and the resulting data was stored for later evaluation. Please let me know if you have any suggestions.

### CODE REVIEW

File: 1.train_models/classify_genotypes/nbconverted/classify_genotypes.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+plate3pdf["Metadata_Plate"] = "Plate_3p"
+plate3pdf = process_plates(plate3pdf)
+p3p_wells = ["F11", "G11", "C3", "F3"]
+rest3pdf, test3pdf = create_splits(p3p_wells, plate3pdf)
+rest3pdf, test3pdf = down_sample_by_genotype(rest3pdf), down_sample_by_genotype(test3pdf)
+
+plate4df["Metadata_siRNA"].fillna("No Construct", inplace=True)
+plate4df.dropna(inplace=True)
+plate4df = plate4df.loc[plate4df["Metadata_siRNA"] == "No Construct"]
+plate4df["Metadata_Plate"] = "Plate_4"
```
Comment (jenna-tomkinson) [inline 1.train_models/classify_genotypes/nbconverted/classify_genotypes.py]: Recommend adding code comments here regarding that you are only including control wells from Plate 4 in the training and testing splits. As well, a comment on why you needed to add Plate_4 in the Metadata_Plate column, is this due to it being missing? If so, please add an issue to the image analysis repo so I possibly can make a fix for it if necessary.

File: 1.train_models/classify_genotypes/nbconverted/classify_genotypes.py [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+
+# ### Input
+
+# In[3]:
+
+
+plate5df_path = pathlib.Path(root_dir / "nf1_painting_repo/3.processing_features/data/single_cell_profiles/Plate_5_sc_normalized.parquet").resolve(strict=True)
+plate3df_path = pathlib.Path(root_dir / "nf1_painting_repo/3.processing_features/data/single_cell_profiles/Plate_3_sc_normalized.parquet").resolve(strict=True)
+plate3pdf_path = pathlib.Path(root_dir / "nf1_painting_repo/3.processing_features/data/single_cell_profiles/Plate_3_prime_sc_normalized.parquet").resolve(strict=True)
+plate4df_path = pathlib.Path(root_dir / "nf1_painting_repo/3.processing_features/data/single_cell_profiles/Plate_4_sc_feature_selected.parquet").resolve(strict=True)
```
Comment (jenna-tomkinson) [inline 1.train_models/classify_genotypes/nbconverted/classify_genotypes.py]: Why do you only use the feature-selected parquet file for Plate_4 but not the rest of the plates? For the rest of the plates, you use the normalized parquet files. Is there a specific reason for this?

File: 1.train_models/classify_genotypes/nbconverted/classify_genotypes.py [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+
+plate5df = pd.read_parquet(plate5df_path)
+plate4df = pd.read_parquet(plate4df_path)
+plate3df = pd.read_parquet(plate3df_path)
+plate3pdf = pd.read_parquet(plate3pdf_path)
+
+rng = np.random.default_rng(0)
```
Comment (jenna-tomkinson) [inline 1.train_models/classify_genotypes/nbconverted/classify_genotypes.py:68]: Recommend including a code comment on what this is for. Is this the same as setting a random seed for reproducibility?

File: 1.train_models/classify_genotypes/nbconverted/classify_genotypes.py [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+        Cleaned single cell data by removing nans and HET cells.
+    """
+
+    _df.dropna(inplace=True)
+    _df = _df.loc[_df[gene_column] != "HET"]
+    return _df
+
+def shuffle_data(_X):
+    """
+    Shuffle the columns of the input dataframe independently.
```
Comment (jenna-tomkinson) [inline 1.train_models/classify_genotypes/nbconverted/classify_genotypes.py:129]: Recommend keeping the code strings similar across functions, e.g., include a short summary of the function for all function not just shuffle_data. Totally optional.

File: 1.train_models/classify_genotypes/nbconverted/classify_genotypes.py [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+plate4df.dropna(inplace=True)
+plate4df = plate4df.loc[plate4df["Metadata_siRNA"] == "No Construct"]
+plate4df["Metadata_Plate"] = "Plate_4"
+plate4df = process_plates(plate4df)
+p4_wells = ["D5", "C11"]
+rest4df, test4df = create_splits(p4_wells, plate4df)
+rest4df, test4df = down_sample_by_genotype(rest4df), down_sample_by_genotype(test4df)
+
+
+# ## Harmonize data across plates to each data split
```
Comment (jenna-tomkinson) [inline 1.train_models/classify_genotypes/nbconverted/classify_genotypes.py]: Recommend not using the term "Harmonize" if you are not using a batch correction method. It looks like below that you are just taking the union of the features across all the plates and then concat all data together based on the split.

File: 1.train_models/classify_genotypes/nbconverted/classify_genotypes.py (hunk 2 above)
Comment (jenna-tomkinson) [reply 1.train_models/classify_genotypes/nbconverted/classify_genotypes.py]: I see you state above, "The feature selected data is used in all data splits.", so I am assuming you want to have all plates with feature selected data not just Plate_4.

**Review by jenna-tomkinson — CHANGES_REQUESTED** [review]
Super exciting to finally get to a final model! I am requesting changes as I see in the `classify_genotypes.py` that you only use the feature-selected data from Plate 4 and just normalized from the rest of the plates. Based on the title of the notebook, it sounds like you want to use feature-selected data for all plates. Once you confirm that, then I can do a quick re-review and approve!

**Review by jenna-tomkinson — APPROVED** [review]
LGTM! 🎉 Feel free to merge! Just one question, how many features are we training on? I didn't see that in the notebook unless I missed it.

**Comment by MattsonCam** [comment]
> LGTM! 🎉 Feel free to merge! Just one question, how many features are we training on? I didn't see that in the notebook unless I missed it.

There are 845 features used for training. Maybe I'll mention that in the README later

**Comment by MattsonCam** [comment]
Thanks @jenna-tomkinson, merging now!

**Pull request merged**